### PR TITLE
fix jaccoeffs

### PR DESCRIPTION
**Pull request by nickhale**
Add more efficient version of jaccoeffs for chebtech-based chebfuns. 

Also fix a few errors in documentation for related methods.

### CODE REVIEW

**Comment by ajt60gaibb** [comment]
I was interested to know what changed here, anything algorithmic? 
Using Clenshaw-Curtis-Jacobi? 

Could one do `cheb2jac` with `0<=alpha<=1` and `0<=beta<=1' using 
asymptotics and then use banded operators to do integer moves around the Jacobi 
plane?  This would cost `O(N (log N)^2/loglogN)' instead of `O(N^2)`. 
Useful for your fractional derivative stuff, perhaps? Worth it, maybe not?

**Comment by nickhale** [comment]
Yes, the algorithmic change is just the switch to Clenshaw-Curtis-Jacobi. 

I thought the problem with the asymptotic based transforms in the Jacobi case is that we don't have a good error formula?

**Comment by ajt60gaibb** [comment]
Oh, in the range `-1/2<alpha<1/2` and `-1/2<beta<1/2` the error is `2x` the first neglected term. See NIST, just under (18.15.1). 

When we were chatting before, I was speaking about general `alpha` and `beta`.

**Comment by nickhale** [comment]
I have a working code for the US case, but not for general Jacobi. I can imagine it being a bit of a mess to write.. Feel free to have a try!

**Comment by ajt60gaibb** [comment]
Yes, you do!  We need to find a cool application for that.  We can arbitrarily move up and down the diagonal of the Jacobi plane... hmmm...  And you are telling me it is not useful for fractional derivatives?  (Feels like it should be.)  

If you tell me it's worth it, then I will do it. I do not plan to do it as a sick idea of a fun few days (though it would be satisfying at the end). This is why Alex Barnett would tell us to construct the asymptotics numerically... he has a point with the applications he has in mind...

**Comment by nickhale** [comment]
If you want to do fractional derivatives of high degree polynomials, then it could be useful.

The only algorithmic change in this pull request is that `cheb2jac()` is used when computing Jacobi coefficients of a chebtech-based chebfun, so please don't let the possibility of improving `cheb2jac()` hold up this pull request.

**Comment by ajt60gaibb** [comment]
What @nickhale did is very good. It is what should have been done in the first place. I support the big green button being pressed.

**Comment by asgeirbirkis** [comment]
I'm looking at this code and assuming no issues arise, will get it merged today or tomorrow.

File: @chebtech/jaccoeffs.m [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    bet = alp;
+    alp = n;
+    n = length(f);
+end
+
+% Convert from Chebyshev to Jacobi coefficients.
+c = cheb2jac(f.coeffs, alp, bet);
+
+% Truncate / pad to length n:
+if ( nargin > 1 )
```
Comment (asgeirbirkis) [inline @chebtech/jaccoeffs.m]: When would nargin not be greater than 1? We'd get errors above for alp and beta missing otherwise?

Comment (ajt60gaibb) [inline @chebtech/jaccoeffs.m]: The help text says that if `nargin=1`, then it computes Legendre coefficients. The code does not do this. 

I think it should return an error if `nargin<4`.

**Comment by asgeirbirkis** [comment]
@nickhale , please look at the changes I made to the documentation in 3f44451. A lot of the jaccoeffs methods had help text that seemed to be more suited to legcoeffs methods. Please let me know if I was correct in fixing this, and have a look if I overlooked anything else that should have been changed.

Also, cf. the comment I made on the outdated diff above (and the comment from @ajt60gaibb on the issue).

**Comment by nickhale** [comment]
Yes, sorry. Looks like I forgot to clear out a lot of the `legcoeffs()` stuff from the documentation.

The changes all look good to me -- thanks!

**Pull request merged**